src/main/java/seedu/address/security/SecurityManager.java

```diff
@@ -38,34 +41,44 @@ public boolean getAuthentication() {
 
     @Override
     public void login(String username, String password) {
-        if (username.equals(this.username) && password.equals(this.password)) {
-            this.isAuthenticated = true;
-            //Instantiates User in Model manager
-            model.matchUserToPerson(username);
-            //TODO Implement logger
-            //System.out.println("Correct Password");
-            raise(new SuccessfulLoginEvent());
-        } else {
-            //TODO Raise an exception to prompt user to type again
-            //TODO implement logger
-            //System.out.println("Incorrect password");
-            raise(new UnsuccessfulLoginEvent());
+        userlist = appUsers.getAccountCredentials();
+        for (AccountCredential acc : userlist) {
+            if (username.equals(acc.getUserName())) {
+                if (acc.passwordIsValid(password)) {
+                    this.isAuthenticated = true;
+                    logic.matchUserToPerson(username);
+                    raise(new SuccessfulLoginEvent());
+                } else {
+                    //THROW INCORRECT PASSWORD EXCEPTION
+                    incorrectPassWord = true;
+                    raise(new NewResultAvailableEvent("Incorrect Password"));
+                }
+            }
+        }
+        //When username is not in the list
+        if (!incorrectPassWord && !this.isAuthenticated) {
+            //THROW USER NOT FOUND
+            raise(new NewResultAvailableEvent("Username not found"));
         }
+        incorrectPassWord = false;
     }
 
     @Override
     public void logout() {
         this.isAuthenticated = false;
-        //TODO Check below is working?
-        model.clearUser();
+        //Clears instance of User in model to prevent anyone accessing previously logged in user
```

Comment:
logic?

Reply:
Not sure what you're referring to, but im guessing why I don't access model anymore. The reason is simple, because model is already accessible through logic. So I decided to take away model from SecurityManager, so now it only cares and know about logic